pkg/probe/http/http_test.go

```diff
@@ -165,6 +165,55 @@ func TestHTTPProbeChecker(t *testing.T) {
 			health:     probe.Success,
 			accBody:    "User-Agent: kube-probe/",
 		},
+		{
+			handler: headerEchoHandler,
```

Comment:
from test coverage perspective I'm surprised there is no test that validates that there are no "unexpected" headers in the payload. All tests validates that adding or removing specific header works.

I realize it's not relevant to this PR, but maybe you be ok adding this validation?

Reply:
What would you consider an invalid header ? 🤔 Isn't any string a valid header ? 

Or do you mean wrong types - not string ?

Reply:
sorry, I meant that I do not see a test that validates that resulting body doesn't have any additional headers. I would expect that test to fail when you would add `Accept` header by default.